compiler/src/dmd/traits.d

```diff
@@ -1081,56 +1102,59 @@ Expression semanticTraits(TraitsExp e, Scope* sc)
                  e.ident == Id.getVirtualMethods ||
                  e.ident == Id.getOverloads)
         {
-            uint errors = global.errors;
-            Expression eorig = ex;
-            ex = ex.expressionSemantic(scx);
-            if (errors < global.errors)
-                e.error("`%s` cannot be resolved", eorig.toChars());
-
-            if (e.ident == Id.getVirtualFunctions)
+            if (e.ident != Id.getOverloads || dim != 1)
```

Comment:
Use `e.ident == Id.getVirtualFunctions || e.ident == Id.getVirtualMethods` instead, as it's much more robust than the nots. Even better, give the deprecation message earlier and get rid of the getVirtualFunctions case.

Reply:
This is confusing in the diff, because the indentation changes. The deprecation check is unrelated to the `if`: I've rewritten it a bit to avoid `!=` regardless. The single-parameter `getOverloads` gets to skip this entire block because it already gets `f` directly.